graphistry/plugins/sentinel.py

```diff
@@ -0,0 +1,699 @@
+                g = graphistry.configure_sentinel(...)
+
+                query = '''
+                SecurityEvent
+                | where TimeGenerated > ago(1d)
+                | where EventID == 4625  // Failed logon
+                | project TimeGenerated, Account, Computer, IpAddress
+                | take 1000
+                '''
+
+                # Query last 7 days
+                df = g.kql(query, timespan=timedelta(days=7))
+                print(f"Found {len(df)} failed logon events")
+
+        **Example: Get all tables as list**
+            ::
+
+                # Always get a list of all tables
+                dfs = g.kql(query, single_table=False)
+                df = dfs[0]
+
+        **Example: Query with specific time range**
+            ::
+
+                from datetime import datetime, timedelta
+
+                # Query specific time window
+                start = datetime(2024, 1, 1)
+                end = datetime(2024, 1, 7)
+                df = g.kql(query, timespan=(start, end))
+
+        **Example: Multi-table query**
+            ::
+
+                query = '''
+                SecurityEvent | summarize Count=count() by EventID | top 5 by Count;
+                SecurityAlert | take 10
+                '''
+
+                # With single_table=False, returns all tables
+                frames = g.kql(query, single_table=False)
+                events_df = frames[0]
+                alerts_df = frames[1]
+        """
+        results = self._sentinel_query(query, timespan=timespan)
+
+        if not results:
+            if single_table:
+                raise ValueError("Query returned no results")
+            return []
+
+        dfs: List[pd.DataFrame] = []
+
+        for result in results:
+            # Determine if we should unwrap nested data
+            do_unwrap = (
+                unwrap_nested is True
+                or (unwrap_nested is None and _should_unwrap(result))
+            )
+
+            if do_unwrap:
+                try:
+                    df_unwrapped = _unwrap_nested(result)
+                    dfs.append(df_unwrapped)
+                    continue
+                except Exception as exc:
+                    if unwrap_nested is True:
+                        raise RuntimeError(f"Failed to unwrap nested data: {exc}") from exc
+                    # Heuristic miss - fall back to flat table
+                    pass
+
+            # Default: flat table
+            if not result.column_names:
+                # Safety fallback
+                dfs.append(pd.DataFrame(result.data))
+            else:
+                dfs.append(pd.DataFrame(result.data, columns=result.column_names))
+
+        # Auto-unbox single table result if requested
+        if single_table:
+            if len(dfs) > 1:
+                logger.warning(f"Query returned {len(dfs)} tables, returning first table only")
+            return dfs[0]
+
+        return dfs
+
+    def kql_last(
```

Comment:
i recommend removing as conceptual bloat for users: seems unncessary as can be folded into `kql()` as part of timespan param, like if `timespan=...` is an int, autoconvert it to hours